fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/codesystem/FhirCodeSystemPutRequest.java

```diff
@@ -43,18 +42,12 @@ public FhirCodeSystemPutRequest(CodeSystem codeSystem) {
 	public Boolean execute(RepositoryContext context) {
 		checkArgument(codeSystem.getConcepts() == null || codeSystem.getConcepts().size() < CONCEPT_LIMIT, "Maintenance of code systems with more than %d codes is not supported.", CONCEPT_LIMIT);
 		
-		final List<FhirCodeSystemCUDSupport> cudSupport = context.service(RepositoryManager.class)
-			.repositories()
-			.stream()
-			.flatMap(r -> r.optionalService(FhirCodeSystemCUDSupport.class).stream())
-			.limit(1)
-			.collect(Collectors.toList());
-		
+		final Optional<FhirCodeSystemCUDSupport> cudSupport = context.optionalService(FhirCodeSystemCUDSupport.class);
 		if (cudSupport.isEmpty()) {
 			throw new IllegalStateException("FHIR CodeSystem resource creation support is missing.");
```

Comment:
How about throwing an `HTTP 501 Not Implemented` in this case?

Reply:
@apeteri did you apply this requested change? If not, could you please do so, thanks!

Reply:
Done